tripal/tripal
# Tv4g2 issue2041 get colum term id on null during config import

# Bug Fix

### Issue #2041

## Description


This PR changes how the function getColumnTermId functions within the FieldType definitions and especially how NULL $mapping is handled. This variable may be NULL during full site configuration imports using Drupal's [https://www.drupal.org/docs/administering-a-drupal-site/configuration-management/managing-your-sites-configuration](Configuration Management) system.

## Testing?



This functionality can be tested by using the above mentioned Configuration import/export system. For best results, use a new site to make the initial configuration export and yet another new site for importing, in both cases having Tripal fully installed and prepared with General content types loaded.
 1. Use the system to export the full site configuration - `/admin/config/development/configuration/full/export`
 2. On the second site, change a configuration on the site. A common example would be to switch the order of genus and species on the Organism content type page.
 3. Use the above system to re-import your changes on the second newly installed Tripal site, remembering to set the UUID from the first site: `drush cset system.site uuid <uuid> -y` which can be found on the first site using `drush cget system.site uuid`.
 4. The action of importing configurations should not throw any errors described in the initial issue, and your genus and species fields should be reverted to their original order.
